vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -257,10 +259,26 @@ def create_weights(
             else:
                 layer.register_parameter("input_scale", None)
 
+    def dequant_block_fp8_weight(self, layer) -> torch.Tensor:
+        if hasattr(layer, "updated_fp8_weight") and layer.updated_fp8_weight:
+            return layer.weight
+        dequant_weight = hpu_ops.dequant_block_fp8_weight_naive(
+            layer.weight,
+            layer.weight_scale_inv.data,
+            self.quant_config.weight_block_size,
+            original_M=layer.orig_M,
+            original_N=layer.orig_N,
+            do_unpad=True,
+        )
+        return dequant_weight
+
     def process_weights_after_loading(self, layer: Module) -> None:
         # TODO(rob): refactor block quant into separate class.
         if self.block_quant:
             assert self.quant_config.activation_scheme == "dynamic"
+            if current_platform.is_hpu():
+                layer = hpu_ops.fp8_block_linear_postprocess_weights(layer)
```

Comment:
@yiliu30 , I wrapped your original set_weight funcs to vllm-hpu-extension and call it here.
BTW, I saw you put it at create_weight instead of process_weights_after_loading, I think we should put it here since we want to set final loaded weight, right? Please check if this change makes sense to you.